### PR TITLE
pre-commit workflow

### PR DESCRIPTION
**Pull request by lmilbaum**

### CODE REVIEW

**Comment by lmilbaum** [comment]
@Zlopez Would you mind activating the Discussions for this project? It would be so much easier to discuss various topics in context of issues and PRs.

**Comment by Zlopez** [comment]
@lmilbaum I can, I just need to check out how to do that.

**Comment by Zlopez** [comment]
They should now be enabled, it was simple.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+default_language_version:
+    python: python3
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 23.1.0
```
Comment (Zlopez) [inline .pre-commit-config.yaml:6]: Will renovate update these versions?

**Comment by Zlopez** [comment]
Could you try to rebase it now, the other PRs are now merged.

**Comment by Zlopez** [comment]
It seems that the pylint check is still failing.

**Comment by lmilbaum** [comment]
> It seems that the pylint check is still failing.
> 
> But we don't use that in tox, so I'm not sure if we want to introduce it in the CI. If we want, it would be probably good to add it in the tox as well. So it doesn't pass through in the local environment and then fail in CI.

There is still a lot of work to do with pylint. I'll handle it one file at a time. In the meantime this PR should not be merged.

**Comment by Zlopez** [comment]
Not sure what exactly is the issue, but I tried to run the pre-commit locally `tox -e pre-commit` and it just ran py311 tests and nothing more. It seems that it's missing some configuration file in tox.

On Github worker it runs something completely different.

**Comment by lmilbaum** [comment]
> tox -e pre-commit

Can you please share the stdout?

**Comment by lmilbaum** [comment]
This is my stdout:
```
(venv) lmilbaum@Lioras-Air-2 anitya % tox -e pre-commit
pre-commit: recreate env because python changed executable='/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/bin/python3.11'->'/opt/homebrew/Cellar/python@3.11/3.11.4_1/Frameworks/Python.framework/Versions/3.11/bin/python3.11'
pre-commit: remove tox env folder /Users/lmilbaum/repos/github.com/lmilbaum/anitya/.tox/pre-commit
pre-commit: install_deps> python -I -m pip install poetry
.pkg: recreate env because python changed executable='/opt/homebrew/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/bin/python3.11'->'/opt/homebrew/Cellar/python@3.11/3.11.4_1/Frameworks/Python.framework/Versions/3.11/bin/python3.11'
.pkg: remove tox env folder /Users/lmilbaum/repos/github.com/lmilbaum/anitya/.tox/.pkg
.pkg: install_requires> python -I -m pip install 'poetry-core>=1.0.0'
.pkg: _optional_hooks> python /Users/lmilbaum/repos/github.com/lmilbaum/anitya/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: get_requires_for_build_sdist> python /Users/lmilbaum/repos/github.com/lmilbaum/anitya/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: prepare_metadata_for_build_wheel> python /Users/lmilbaum/repos/github.com/lmilbaum/anitya/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: build_sdist> python /Users/lmilbaum/repos/github.com/lmilbaum/anitya/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
pre-commit: install_package_deps> python -I -m pip install 'Flask-Login<0.7.0,>=0.6.2' 'Flask-WTF<2.0.0,>=1.0.1' 'Flask<3.0.0,>=2.1.2' 'Jinja2<3.1.3' 'SQLAlchemy<2.0.0,>=1.4.41' 'WTForms<4.0.0,>=3.0.1' Werkzeug==2.3.6 'alembic<2.0.0,>=1.8.1' 'anitya-schema<3.0.0,>=2.0.1' 'arrow<2.0.0,>=1.2.3' 'beautifulsoup4<5.0.0,>=4.11.1' 'defusedxml<0.8.0,>=0.7.1' 'fedora-messaging<4.0.0,>=3.1.0' 'ordered-set<5.0.0,>=4.1.0' 'packaging<24.0,>=23.0' 'python-dateutil<3.0.0,>=2.8.2' 'semver<4.0.0,>=3.0.0' 'social-auth-app-flask-sqlalchemy<2.0.0,>=1.0.1' 'social-auth-app-flask<2.0.0,>=1.0.0' 'sseclient<0.0.28,>=0.0.27' 'straight.plugin<2.0.0,>=1.5.0' 'toml<0.11.0,>=0.10.2' 'webargs<9.0.0,>=8.2.0'
pre-commit: install_package> python -I -m pip install --force-reinstall --no-deps /Users/lmilbaum/repos/github.com/lmilbaum/anitya/.tox/.tmp/package/144/anitya-1.8.1.tar.gz
pre-commit: commands[0]> pre-commit run --all-files
black....................................................................Passed
flake8...................................................................Passed
isort....................................................................Passed
pylint...................................................................Passed
mypy.....................................................................Passed
renovate-config-validator................................................Passed
.pkg: _exit> python /Users/lmilbaum/repos/github.com/lmilbaum/anitya/venv/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
  pre-commit: OK (75.50=setup[56.43]+cmd[19.06] seconds)
  congratulations :) (75.57 seconds)
(venv) lmilbaum@Lioras-Air-2 anitya % 
```

**Comment by Zlopez** [comment]
Tried it again and it's now failing with exit code 2:
```
tox -e pre-commit                                                                                                                                                                        1.2m  Thu Jun 29 10:50:17 2023
.pkg: _optional_hooks> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: get_requires_for_build_sdist> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: prepare_metadata_for_build_wheel> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: build_sdist> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
pre-commit: install_package> python -I -m pip install --force-reinstall --no-deps /var/home/zlopez/git/anitya/.tox/.tmp/package/97/anitya-1.8.1.tar.gz
pre-commit: commands[0]> pre-commit run --all-files
pre-commit: exit 2 (0.00 seconds) /var/home/zlopez/git/anitya> pre-commit run --all-files
.pkg: _exit> python /usr/lib/python3.11/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
  pre-commit: FAIL code 2 (29.09=setup[29.08]+cmd[0.00] seconds)
  evaluation failed :( (29.20 seconds)
```

My assumption is that the pre-commit is not available at all, it should be added in poetry test requirements.

**Comment by Zlopez** [comment]
As I though, there were a few missing dependencies on my machine, so I added them in https://github.com/fedora-infra/anitya/pull/1646

**Comment by lmilbaum** [comment]
> As I though, there were a few missing dependencies on my machine, so I added them in #1646

Thank you so much

**Pull request merged**